### PR TITLE
Fix #487 - Add permissions to silo edit page.

### LINKED ISSUE #487: Add permission control to silo edit page
### Current behavior
If user try to access silo edit page with url then they can access the page even they don't have a permission on that table.

### Expected behavior
If users try to access silo edit page and if they are not owner or table isn't shared with them, then they can't access the page.

### Steps to reproduce the behavior

1. GIVEN user try to access silo edit page with url (/silo_edit/{id}) 
2. WHEN they haven't any permission with that table
3. THEN they can still access to page.

### PR DESCRIPTION
**Pull request by yilmazbekdemir**
## Purpose
Users can access silo_edit page via url event if they don't have a permission with that silo.

## Approach
Check for Permissions added to silo_edit  view.

_Related ticket: #487

### CODE REVIEW

File: silo/views.py [hunk 1]
```diff
@@ -437,6 +437,23 @@ def editSilo(request, id):
...
+
+    request_user_org=None
+    owner_user_org=None
+    if(hasattr(request.user, 'tola_user') and
+            hasattr(edited_silo.owner,'tola_user')):
+        request_user_org = request.user.tola_user.organization
+        owner_user_org = edited_silo.owner.tola_user.organization
+
+    if not (edited_silo.owner == request.user or edited_silo.public
+            or request.user in edited_silo.shared.all()
```
Comment (jefmoura) [inline silo/views.py]: It better to use the `exists` method instead of fetching the whole data from the db.

**Comment by yilmazbekdemir** [comment]
I think we should change all functional views about silo, Silo detail, edit and list should also change. I think we can spend some times to moving them to class-based views and adding PEP8 check to entire project.

**Comment by jefmoura** [comment]
My concern about updating those things now is that we will change Track to be only a Rest API

File: silo/views.py [hunk 2]
```diff
@@ -430,14 +430,29 @@ def appendTwoSilos(mapping_data, lsid, rsid, msid):
...
+
+    request_user_org = None
+    owner_user_org = None
+    if(hasattr(request.user, 'tola_user') and
+       hasattr(edited_silo.owner,'tola_user')):
+        request_user_org = request.user.tola_user.organization
+        owner_user_org = edited_silo.owner.tola_user.organization
+
+    if not (edited_silo.owner == request.user or edited_silo.public
+            or Silo.objects.filter(pk=id, shared__id=request.user.pk).exists()
```
Comment (jefmoura) [inline silo/views.py]: Create a variable for that. Don't put this logic inside of a conditional.

File: silo/tests/test_views.py [hunk 3]
```diff
@@ -1307,4 +1307,109 @@ def test_list_public_silo(self):
         request.user = request_user
         response = views.list_silos(request)
         self.assertEqual(response.status_code, 200)
-        self.assertContains(response, 'Test Share Silo')
\ No newline at end of file
+        self.assertContains(response, 'Test Share Silo')
+
+
+class SiloEditViewTest(TestCase):
```
Comment (jefmoura) [inline silo/tests/test_views.py]: Add a test when the table is `public`.

File: silo/views.py [hunk 4]
```diff
@@ -443,7 +443,7 @@ def editSilo(request, id):
     request_user_org = None
     owner_user_org = None
     if(hasattr(request.user, 'tola_user') and
-            hasattr(edited_silo.owner,'tola_user')):
+       hasattr(edited_silo.owner,'tola_user')):
```
Comment (jefmoura) [inline silo/views.py]: Missing whitespace after ','

File: silo/tests/test_views.py [hunk 5]
```diff
@@ -1335,20 +1334,13 @@ def test_silo_edit_page_with_unauthorized_user(self, mock_get_by_url,
...
         request.user = request_user
-        request.session = 'session'
-        message_storage = FallbackStorage(request)
-        request._messages = message_storage
-        views.silo_detail(request, silo.pk)
-        messages = []
-        for m in message_storage:
-            messages.append(m.message)
+        response = views.edit_silo(request, silo.pk)
+        self.assertEqual(response.status_code, 404)
```
Comment (jefmoura) [inline silo/tests/test_views.py]: This isn't PEP8, you're changing the logic of the tests.

File: silo/tests/test_views.py [hunk 6]
```diff
@@ -1307,4 +1307,147 @@ def test_list_public_silo(self):
...
+                              reads=[read],
+                              public=True,
+                              shared=[],
+                              share_with_organization=False)
+
+        url = reverse('edit_silo', args=[silo.pk])
+
+        request = self.factory.get(url)
+        request.user = request_user
+        response = views.edit_silo(request, silo.pk)
```
Comment (jefmoura) [inline silo/tests/test_views.py]: The view isn't named `edit_silo` in this commit.

**Pull request merged**